[SOAR-1904][SOAR-1903][SOAR-1990] - Insight agent plugin (#585)
rapid7/insightconnect-plugins: rapid7_insight_agent/help.md

# Description

> Take actions on an Insight Agent

Reviewer: Improve this with more details about the Insight Agent, there's existing docs we can pull from.

Have links too

# Key Features

> Identify key features of plugin.

Reviewer: Need to fill out Key Features and Requirements

# Requirements

* Example: Requires an API Key from the product
* Example: API must be enabled on the Settings page in the product's user interface

# Documentation
## Setup

The connection configuration accepts the following parameters:

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|api_key|credential_secret_key|None|False|X API Key from the Insight Platform|None|a5zy0a6g-504e-46bz-84xx-1b3f5ci36l99|
|org_id|credential_secret_key|None|False|Organization ID|None|2cg99z6y-795n-7bzf-hj67-12355h554974|
|region|string|United States|False|Region|['United States', 'Europe', 'Canada', 'Australia', 'Japan']|United States|

Example input:

```
{
  "api_key": "a5zy0a6g-504e-46bz-84xx-1b3f5ci36l99",
  "org_id": "2cg99z6y-795n-7bzf-hj67-12355h554974",
  "region": "United States"
}

> ```

Reviewer: This needs to pass mdl - needs a line break after

Example input:

```
{
  "agent": "MaliciousHost"
}
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|agent|agent|True|Agent information|

Example output:

```
{
  "agent": {
    "id": "15eec979a15f75ad70567d58ad8a0aef",
    "vendor": "Microsoft",
    "version": "SP1",
    "description": "Microsoft Windows 7 Professional SP1",
    "hostNames": [
      {
        "name": "joey-w7-vagrant-pc"
      }
    ],
    "primaryAddress": {
      "ip": "10.0.2.15",
      "mac": "08-00-27-33-9F-CE"
    },
    "uniqueIdentity": [
      {
        "source": "Endpoint Agent",
        "id": "15eec979a15f75ad70567d58ad8a0aef"
      },
      {
        "source": "CSPRODUCT",
        "id": "030B19C0-F12D-6643-B86E-FA7C08A4A838"
      }
    ],
    "attributes": [
      {
        "key": "cpuinfo",
        "value": "Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz"
      },
      {
        "key": "timezone",
        "value": "GMT-7"
      },
      {
        "key": "release",
        "value": "7"
      },
      {
        "key": "proxies",
        "value": "{}"
      }
    ]
  }
}
```

### Triggers

_This plugin does not contain any triggers._

### Custom Output Types

> _This plugin does not contain any custom output types._

Reviewer: New line after this

## Troubleshooting

_This plugin does not contain any troubleshooting information._

# Version History

* 1.0.0 - Initial plugin

# Links

## References

> * [Rapid7 Insight Agent](LINK TO PRODUCT/VENDOR WEBSITE)

Reviewer: Add links to Agent documetnation

# Description

> Using the power of the [Insight Agent](https://docs.rapid7.com/insight-agent/overview/) investigate and mitigate vulnerabilities across your organization. 

Reviewer 2: ```suggestion
Investigate and mitigate vulnerabilities across your organization using the power of the [Insight Agent](https://docs.rapid7.com/insight-agent/overview/). 
```

Reads better IMO

# Key Features

* Get details about devices

> * Qurantine and unquarantine devices

Reviewer 2: ```suggestion
* Quarantine and unquarantine devices
```

# Requirements

> * Platform API Key: [Mange Platform API Keys](https://docs.rapid7.com/insight/managing-platform-api-keys/)

Reviewer 2: ```suggestion
* [Platform API Key](https://docs.rapid7.com/insight/managing-platform-api-keys/)
```

> * Organization ID: (See Page 10 of the GraphQL API doc)

Reviewer 2: Can you link the page here?

Reviewer: I don' think the GraphQL API doc is released to customers at this point. We should document more here if we can do it with brevity

Author: I didn't know what to do with this...getting the org ID is rough. I had to pull it from the network traffic in chrome.

Reviewer: Is this Org ID different the one in the browser URL when you log into the platform?

e..g https://us.connect.insight.rapid7.com/**ABCDEFGHIJKLMNOP123**#/home

    "version": "SP1",
    "description": "Microsoft Windows 7 Professional SP1",
    "hostNames": [
      {
        "name": "joey-w7-vagrant-pc"
      }
    ],
    "primaryAddress": {
      "ip": "10.0.2.15",
      "mac": "08-00-27-33-9F-CE"
    },
    "uniqueIdentity": [
      {
        "source": "Endpoint Agent",
        "id": "15eec979a15f75ad70567d58ad8a0aef"
      },
      {
        "source": "CSPRODUCT",
        "id": "030B19C0-F12D-6643-B86E-FA7C08A4A838"
      }
    ],
    "attributes": [
      {
        "key": "cpuinfo",
        "value": "Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz"
      },
      {
        "key": "timezone",
        "value": "GMT-7"
      },
      {
        "key": "release",
        "value": "7"
      },
      {
        "key": "proxies",
        "value": "{}"
      }
    ]
  }
}
```

### Triggers

_This plugin does not contain any triggers._

### Custom Output Types

_This plugin does not contain any custom output types._

## Troubleshooting

_This plugin does not contain any troubleshooting information._

# Version History

* 1.0.0 - Initial plugin

# Links

## References

* [Rapid7 Insight Agent](https://docs.rapid7.com/insight-agent/overview/)

> * [Mange Platform API Keys](https://docs.rapid7.com/insight/managing-platform-api-keys/)

Reviewer 2: ```suggestion
* [Manage Platform API Keys](https://docs.rapid7.com/insight/managing-platform-api-keys/)
```

# Description

> The [Insight Agent] (https://docs.rapid7.com/insight-agent/overview/) collects live system information, easily centralizing and monitoring data on the Insight platform. Using the Insight Agent plugin from InsightConnect, you can quarantine, unquarantine and monitor potentially malicious IPs, addresses, hostnames, and devices across your organization. 

Reviewer: ```suggestion
The [Insight Agent](https://docs.rapid7.com/insight-agent/overview/) collects live system information, easily centralizing and monitoring data on the Insight platform. Using the Insight Agent plugin from InsightConnect, you can quarantine, unquarantine and monitor potentially malicious IPs, addresses, hostnames, and devices across your organization. 
```

# Description

The [Insight Agent](https://docs.rapid7.com/insight-agent/overview/) collects live system information, easily centralizing and monitoring data on the Insight platform. Using the Insight Agent plugin from InsightConnect, you can quarantine, unquarantine and monitor potentially malicious IPs, addresses, hostnames, and devices across your organization. 

> The agent is used by Rapid7 InsightIDR and InsightVM customers to monitor endpoints.

Reviewer: ```suggestion
The agent is used by [Rapid7 InsightIDR](https://www.rapid7.com/products/insightidr/) and [InsightVM](https://www.rapid7.com/products/insightvm/) customers to monitor endpoints.
```

    "hostNames": [
      {
        "name": "joey-w7-vagrant-pc"
      }
    ],
    "primaryAddress": {
      "ip": "10.0.2.15",
      "mac": "08-00-27-33-9F-CE"
    },
    "uniqueIdentity": [
      {
        "source": "Endpoint Agent",
        "id": "15eec979a15f75ad70567d58ad8a0aef"
      },
      {
        "source": "CSPRODUCT",
        "id": "030B19C0-F12D-6643-B86E-FA7C08A4A838"
      }
    ],
    "attributes": [
      {
        "key": "cpuinfo",
        "value": "Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz"
      },
      {
        "key": "timezone",
        "value": "GMT-7"
      },
      {
        "key": "release",
        "value": "7"
      },
      {
        "key": "proxies",
        "value": "{}"
      }
    ]
  }
}
```

### Triggers

_This plugin does not contain any triggers._

### Custom Output Types

_This plugin does not contain any custom output types._

## Troubleshooting

_This plugin does not contain any troubleshooting information._

# Version History

* 1.0.0 - Initial plugin

# Links

## References

* [Rapid7 Insight Agent](https://docs.rapid7.com/insight-agent/overview/)
* [Manage Platform API Keys](https://docs.rapid7.com/insight/managing-platform-api-keys/)

> *  

Reviewer: ```suggestion
```

# Description

> Investigate and mitigate vulnerabilities across your organization using the power of the [Insight Agent](https://docs.rapid7.com/insight-agent/overview/). 

Reviewer 3: ```suggestion
The [Insight Agent] (https://docs.rapid7.com/insight-agent/overview/) collects live system information, easily centralizing and monitoring data on the Insight platform. Using the Insight Agent plugin from InsightConnect, you can quarantine, unquarantine and monitor potentially malicious IPs, addresses, hostnames, and devices across your organization. 
```
